dali/test/datatype_convertions.h

```diff
@@ -0,0 +1,37 @@
+#ifndef DALI_DATATYPE_CONVERTIONS_H
+#define DALI_DATATYPE_CONVERTIONS_H
+
+#include "dali/pipeline/data/tensor_list.h"
+
+namespace dali {
+
+namespace detail {
+
+template<typename Backend, typename Input>
```

Comment:
What I'd like to see here would be functions that convert vectors, cv::Mat and vectors thereof.
E.g.
```
template <typename Backend, typename T>
TensorList<Backend> ToTensorList(const std::vector<T> &items);  // expands each item in "items" into a tensor
// automatically expand nested vectors/arrays to tensors

template <typename Backend, typename T, typename integer>
TensorList<Backend> ToTensorList(const std::vector<T> &items, const std::vector<integer> &shape);

template <typename Backend, typename T>
TensorList<Backend> ToTensorList(const std::vector<cv::Mat> &images, DALIImageType type, DALITensorLayout layout);
```

Reply:
If I understood you correctly, you wanted to point out, that in this case `ToTensorList` in insecure, meaning I could put there any type I want?